### ISSUE TITLE
Getting "Path too long" when running the HOL

### ISSUE THREAD

**Issue by dburnison** [Microsoft/PartsUnlimited#55]
I am following the lab. When I get to "4: Create a Release Definition / 2. Test the ARM Template Deployment" I start a release and it gets to the "Azure Deployment:Create Or Update Resource Group action on Parts Unlimited" step and it fails. The error is "##[error]The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.". I have attached the log files (as text files).

Dave Burnison
Sr. ALM Consultant
Microsoft - Premier Developer

[1_Release_log.txt](https://github.com/Microsoft/PartsUnlimited/files/137072/1_Release_log.txt)
[2_Download artifacts_log.txt](https://github.com/Microsoft/PartsUnlimited/files/137073/2_Download.artifacts_log.txt)
[1_Initialize_log.txt](https://github.com/Microsoft/PartsUnlimited/files/137074/1_Initialize_log.txt)

**Comment by EMaher** [comment]
Dave,

The instructions for that lab were just (as in 5 minutes ago) updated.  I think the updates will help you with the issue you are seeing.  Could you give the HOL another try and let us know?

Thanks,
Elizabeth

**Comment by dburnison** [comment]
Thanks Elizabeth. I will give this a try in the morning. 


Dave Burnison
DaveBurnison@yahoo.com 

Sent from Outlook for iPhone

    _____________________________

**Comment by dburnison** [comment]
[1_Initialize_log.txt](https://github.com/Microsoft/PartsUnlimited/files/141550/1_Initialize_log.txt)
[1_Release_log.txt](https://github.com/Microsoft/PartsUnlimited/files/141549/1_Release_log.txt)
[2_Download artifacts_log.txt](https://github.com/Microsoft/PartsUnlimited/files/141551/2_Download.artifacts_log.txt)


Elizabeth,

I went through the lab again and updated the things that looked like had changed and tried to test this again but, got the same error. 

##[error]The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.

I have attached the log files. 

Should I have started from scratch instead of updating the build and release that I used originally?

If needed, I can work with someone on Tuesday to troubleshoot this issue.

**Comment by dtzar** [comment]
Sorry - these are more issues due to the invalid path. You should use $(Build.StagingDirectory) instead of anywhere you see "C:\Agent\HOL".  Hope to have this fix into the HOL very soon.

**Comment by dtzar** [comment]
This has now been updated https://github.com/Microsoft/PartsUnlimited/blob/hands-on-labs/docs/HOL_PartsUnlimited_WebSite_Continuous_Integration/HOL_PartsUnlimited_WebSite_Continuous_Integration_with_Visual_Studio_Online_Build.md Can you please see if you still have problems?

**Comment by dtzar** [comment]
@dburnison Are you still getting this issue?  Open to help figure out what's going on.

**Comment by dburnison** [comment]
Sorry for not responding sooner. I was able to get past the issue. Part of it was my own doing as I had entered some data in the wrong field when configuring the Azure Resource Group task. The updates that were made to the lab did help make things more clear. I did find a couple of issues such as didn't realize that the Resource Group Name cannot contain any spaces and I had to select "Azure Resource Manager" rather than "Azure" when creating the service endpoint.
